.github/workflows/tests-outerloop.yml

```diff
@@ -61,42 +61,80 @@ jobs:
+        if: always()
+        uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
+        with:
+          name: logs-${{ matrix.tests.project }}-${{ matrix.tests.os }}
+          path: |
+            ${{ github.workspace }}/artifacts/log/*/*.binlog
+            ${{ github.workspace }}/artifacts/log/*/TestLogs/**
+            ${{ github.workspace }}/artifacts/TestResults/*/*.trx
+          retention-days: 5
+
+      - name: Upload trx reports
+        if: always()
+        uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
+        with:
+          name: trx-${{ matrix.tests.project }}-${{ matrix.tests.os }}
+          path: |
+            ${{ github.workspace }}/artifacts/log/*/TestLogs/**/*.log
+            ${{ github.workspace }}/artifacts/TestResults/*/*.trx
+          retention-days: 30
+
+  results:
+    if: always()
+    runs-on: ubuntu-latest
+    name: Final Results
+    needs: run_tests
+    steps:
+      # get all the test-job-result* artifacts into a single directory
+      - uses: actions/download-artifact@cc203385981b70ca67e1cc392babf9cc229d5806 # v4.1.9
+        with:
+          pattern: trx-*
+          path: trx-reports
+
       - name: Process logs and post results
         if: always()
         shell: pwsh
         run: |
-          $logDirectory = "${{ github.workspace }}/artifacts/TestResults"
+          $logDirectory = "trx-reports"
           $trxFiles = Get-ChildItem -Path $logDirectory -Filter *.trx -Recurse
 
           $testResults = @() # Initialize an array to store test results
 
           foreach ($trxFile in $trxFiles) {
               # Load the .trx file as XML
-              $xmlContent = [xml](Get-Content -Path $trxFile.FullName)
-
-              # Extract test results from the XML
-              foreach ($testResult in $xmlContent.TestRun.Results.UnitTestResult) {
-                  $testName = $testResult.testName
-                  $outcome = $testResult.outcome
-                  $duration = $testResult.duration
-
-                  # Map outcome to emoji
-                  switch ($outcome) {
-                      "Passed" { $emoji = "✔️" }
-                      "Failed" { $emoji = "❌" }
-                      default { $emoji = "❔" }
-                  }
-
-                  # Normalize the duration to a consistent format (hh:mm:ss.fff)
-                  $normalizedDuration = [TimeSpan]::Parse($duration).ToString("mm\:ss\.fff")
-
-                  # Add the test result to the array
-                  $testResults += [PSCustomObject]@{
-                      TestName    = $testName
-                      Outcome     = $outcome
-                      OutcomeIcon = $emoji
-                      Duration    = $normalizedDuration
+              try {
+                  # Attempt to load the .trx file as XML
+                  $xmlContent = [xml](Get-Content -Path $trxFile.FullName)
+
+                  # Extract test results from the XML
```

Comment:
![image](https://github.com/user-attachments/assets/3c0de906-aa22-4313-a9a1-cb7be87f1d4d)

🤔 We should de-dup successful runs - if a test passed on both OS, then it passed. However, when a test failed on a specific OS we'd want to know that, right? But the current repot isn't showing this info.

Reply:
I would prefer to add the OS as another column. And not do the deduping at all. Deduping so some tests have one entry but others have two, would be confusing. And we don't really save anythingwith that. But your idea for showing the OS is good 👍